src/client-side-encryption/auto_encrypter.ts

```diff
@@ -237,6 +245,16 @@ export class AutoEncrypter {
     this._proxyOptions = options.proxyOptions || {};
     this._tlsOptions = options.tlsOptions || {};
     this._kmsProviders = options.kmsProviders || {};
+    this._credentialProviders = options.credentialProviders;
+
+    if (
+      options.credentialProviders?.aws &&
+      !isEmptyCredentials('aws', options.kmsProviders || {})
```

Comment:
```suggestion
      !isEmptyCredentials('aws', this._kmsProviders)
```
(optional) - kmsProviders is already defaulted to `{}` when assigning to `this._kmsProviders`.

Reply:
Updated.